### DISCUSSION TITLE
Are Containers rendering input elements with a different name supported?

### DISCUSSION THREAD

**Discussion by mdesantis** [react-hook-form/react-hook-form#7361] [Q&A]
I wonder if `Container`s rendering input elements with a different `name` are supported. My use case is servers requiring specific `name` namings, which conflict with this library constraints: for example, Ruby on Rails requires input elements with the same `name` in order to receive arrays of objects:

```html
<input type="hidden" name="social_account[handled_errors][][attribute]" value="name" />
<input type="hidden" name="social_account[handled_errors][][type]" value="blank" />
<input type="hidden" name="social_account[handled_errors][][attribute]" value="app_id" />
<input type="hidden" name="social_account[handled_errors][][type]" value="blank" />
```

Unfortunately, client side form handling libraries are pretty choosy about `name` attribute naming too: for example, `react-hook-form` doesn't allow input elements with the same `name`. But `Controller` lets you specify a different name for its input element, which is something very handy in this case:

```jsx
<Controller
  name="firstName"
  control={control}
  defaultValue=""
  className="materialUIInput"
  render={({ field: { onBlur, onChange, ref, value } }) => {
    return (
      <TextField
        name="users[first_name]"
        variant="outlined"
        onChange={onChange}
        onBlur={onBlur}
        inputRef={ref}
        value={value}
      />
    );
  }}
/>
```

I have even a fully working example here: https://codesandbox.io/s/react-hook-form-with-ui-library-forked-5mfzb

My concerns are: can I rely on this feature, or is it something not supported that can stop working the next minor release without notice? Should I expect weird issues? Should I give up using this library if I need unsupported `name` setups?

Thank you for this awesome library!

**Comment by bluebill1049** [comment]
This is valid for `Controller` or `useController`.
